python/pyarrow/tests/parquet/test_basic.py

```diff
@@ -352,6 +352,37 @@ def test_byte_stream_split(use_legacy_dataset):
                          use_legacy_dataset=use_legacy_dataset)
 
 
+@parametrize_legacy_dataset
+def test_col_encoding(use_legacy_dataset):
+    arr_float = pa.array(list(map(float, range(100))))
+    arr_int = pa.array(list(map(int, range(100))))
+    mixed_table = pa.Table.from_arrays([arr_float, arr_int],
+                                       names=['a', 'b'])
+
+    # Check NONE col_encoding.
+    _check_roundtrip(mixed_table, expected=mixed_table, use_dictionary=False,
+                     col_encoding=None, use_legacy_dataset=use_legacy_dataset)
+
+    # Check "BYTE_STREAM_SPLIT" for column 'a' and "PLAIN" col_encoding for
+    # column 'b'.
+    _check_roundtrip(mixed_table, expected=mixed_table, use_dictionary=False,
+                     col_encoding={'a': "BYTE_STREAM_SPLIT", 'b': "PLAIN"},
+                     use_legacy_dataset=use_legacy_dataset)
+
+    # Check "RLE" for column 'a' and "BYTE_STREAM_SPLIT" col_encoding for
+    # column 'b'.
+    _check_roundtrip(mixed_table, expected=mixed_table,
+                     use_byte_stream_split=['a'],
+                     col_encoding={'a': "RLE", 'b': "BYTE_STREAM_SPLIT"},
+                     use_legacy_dataset=use_legacy_dataset)
+
+    # Check "DELTA_BINARY_PACKED" col_encoding for column 'b'.
+    _check_roundtrip(mixed_table, expected=mixed_table,
+                     use_dictionary=['b'],
+                     col_encoding={'b': "DELTA_BINARY_PACKED"},
+                     use_legacy_dataset=use_legacy_dataset)
+
```

Comment:
Can you add tests for some failure cases (asserting they raise a proper error with `pytest.raises`, see the test above or below): 1) passing an unknown encoding in the dict, 2) passing an unknown column name in the dict, and 3) not passing

Reply:
Passing an unknown column name in the dict currently doesn't give an error (same in `use_byte_stream_split`). Not sure how to add this check in General writer properties - `_create_writer_properties()`. Also the item 3) not passing isn't clear to me. Do you mean if `column_encoding` isn't specified in the `write_table`?

Reply:
> Passing an unknown column name in the dict currently doesn't give an error (same in use_byte_stream_split). Not sure how to add this check in General writer properties 

See the other comment I just added

> Also the item 3) not passing isn't clear to me. Do you mean if column_encoding isn't specified in the write_table?

Ah, I think I didn't finish my sentence there :) I suppose I meant "not passing a dict" (but eg a single string).